File: unit/calculadora_go/cmd/calculadora/calculadora.go
package main

type Calculadora struct {
	historico []int
}

func (c Calculadora) Somar(a, b int) int {
	resultado := a + b
	c.adicionarAoHistorico(resultado)
	return resultado
}

func (c Calculadora) Subtrair(a, b int) int {
	resultado := a - b
	c.adicionarAoHistorico(resultado)
	return resultado
}

func (c Calculadora) Multiplicar(a, b int) int {
	resultado := a * b
	c.adicionarAoHistorico(resultado)
	return resultado
}

func (c Calculadora) Dividir(a, b int) int {
	resultado := a / b
	c.adicionarAoHistorico(resultado)
	return resultado
}

func (c Calculadora) adicionarAoHistorico(a int) {
	c.historico = append(c.historico, a)
}

func (c Calculadora) lerHistorico() ([]int) {
	return c.historico
}

File: unit-io/calculadora_java/src/main/calc/Calculadora.java
package calc;

import java.util.ArrayList;

public class Calculadora {
	private ArmazenamentoArquivo historico;
	
	public Calculadora(ArmazenamentoArquivo arm) {
		this.historico = arm;
	}
	
	public int somar(int a, int b) {
		int resultado = a + b; 
		this.historico.registrar(resultado);
		return resultado;
	}
	
	public int subtrair(int a, int b) {
		int resultado = a - b;
		this.historico.registrar(resultado);
		return resultado;
	}
	
	public int multiplicar(int a, int b) {
		int resultado = a * b;
		this.historico.registrar(resultado);
		return resultado;
	}
	
	public int dividir(int a, int b) {
		int resultado = a / b;
		this.historico.registrar(resultado);
		return resultado;
	}
	
	public ArrayList<Integer> lerHistorico() {
		return this.historico.lerHistorico();
	}
}
